internal/config/config.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/project"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/yaml"
+)
+
+const DefaultPath = "PROJECT"
+
+// Config is the unmarshalled representation of the configuration file
+type Config struct {
+	// Version is the project version, defaults to "1" (backwards compatibility)
+	Version string `json:"version,omitempty"`
+
+	// Domain is the domain associated with the project and used for API groups
+	Domain string `json:"domain,omitempty"`
+
+	// Repo is the go package name of the project root
+	Repo string `json:"repo,omitempty"`
+
+	// Resources tracks scaffolded resources in the project
+	// This info is tracked only in project with version 2
+	Resources []Resource `json:"resources,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+// Check verifies that the configuration file exists
+func Check(path string) (bool, error) {
```

Comment:
```suggestion
func CheckProjectFile() (bool, error) {
      return check(DefaultPath) (bool, error) {
}
func check(path string) (bool, error) {
```

Reply:
https://github.com/kubernetes-sigs/kubebuilder/pull/1325#discussion_r368219213 reasoning for not using "project" and "file" applies here. Remember that this method is called `config.Check()`.

I will do the default path function though.

Reply:
Is it checking the Config structure or the existence and parse of the File?

Reply:
It is checking if the project is configured, and most of the logic is already in the package you mention (`IsProjectConfigured`, `DieIfNotConfigured`, ...). The part in `config` is just the straight lookup of the file.

Maybe `config.Check` is not the best name, will use `config.Exists`.